src/tests/utests/xi_utest_sft_logic.c

```diff
@@ -231,6 +231,292 @@ XI_TT_TESTCASE_WITH_SETUP(
         xi_sft_free_context( &sft_context );
     } )
 
+XI_TT_TESTCASE_WITH_SETUP( xi_utest__select_next_resource_to_download__null_context,
+                           xi_utest_setup_basic,
+                           xi_utest_teardown_basic,
+                           NULL,
+                           {
+                               xi_state_t state =
+                                   _xi_sft_select_next_resource_to_download( NULL );
+
+                               tt_want_int_op( XI_INVALID_PARAMETER, ==, state );
+                           } )
+
+XI_TT_TESTCASE_WITH_SETUP(
+    xi_utest__select_next_resource_to_download__picks_first_element,
+    xi_utest_setup_basic,
+    xi_utest_teardown_basic,
+    NULL,
+    {
+        xi_control_message_file_desc_ext_t three_file_list[3];
+        memset( three_file_list, 0, 3 * sizeof( xi_control_message_file_desc_ext_t ) );
+
+        xi_control_message_t file_update_available = {
+            .file_update_available = {
+                .common = {.msgtype = XI_CONTROL_MESSAGE_SC__SFT_FILE_UPDATE_AVAILABLE,
+                           .msgver  = 1},
+                .list_len = 3,
+                .list     = three_file_list}};
+
+        int32_t download_order[] = {0, 1, 2};
+        xi_sft_context_t sft_context;
+        sft_context.updateable_files_download_order = download_order;
+        sft_context.update_current_file             = NULL;
+        sft_context.update_message_fua              = &file_update_available;
+
+        xi_state_t state = _xi_sft_select_next_resource_to_download( &sft_context );
+        tt_want_int_op( XI_STATE_OK, ==, state );
+        tt_want_ptr_op( sft_context.update_current_file, ==, &three_file_list[0] );
+    } )
+
+XI_TT_TESTCASE_WITH_SETUP(
+    xi_utest__select_next_resource_to_download__negative_index_in_download_order,
+    xi_utest_setup_basic,
+    xi_utest_teardown_basic,
+    NULL,
+    {
+        // ARRANGE
+        xi_control_message_file_desc_ext_t three_file_list[3];
```

Comment:
maybe you could cut this arrange to a separate function since these lines appear multiple times in most of the test cases. but I leave it up to you, both have pros and cons.